processout-sdk/src/main/java/com/processout/processout_sdk/ProcessOut.java

```diff
@@ -233,20 +242,103 @@ public void error() {
                     }
                 });
                 break;
+            case REDIRECT:
             case URL:
+                // Create external webview
                 WebView theWebPage = new WebView(this.context);
                 theWebPage.getSettings().setJavaScriptEnabled(true);
                 theWebPage.getSettings().setPluginState(WebSettings.PluginState.ON);
                 Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(cA.getValue()));
                 this.context.startActivity(browserIntent);
                 break;
+            case FINGERPRINT:
+                // Build the fingerprint hidden webview
+                final WebView fingerPrintWebView = new WebView(this.context);
+                fingerPrintWebView.getSettings().setJavaScriptEnabled(true); // enable javascript
+                fingerPrintWebView.setVisibility(View.INVISIBLE); // Hide the Webview
+
+                // Setup the timeout handler
+                final Handler timeOutHandler = new android.os.Handler();
+                final Runnable timeoutClearer = new Runnable() {
+                    @Override
+                    public void run() {
+                        // Preparing the webview removal
+                        fingerPrintWebView.removeAllViews();
+                        fingerPrintWebView.clearHistory();
+                        fingerPrintWebView.clearCache(true);
+                        fingerPrintWebView.onPause();
+                        // Removing the webview and returning the error
+                        ((ViewGroup) fingerPrintWebView.getParent()).removeView(fingerPrintWebView);
+                        fingerPrintWebView.destroy();
+                        handler.onError(new ProcessOutNetworkException("Fingerprint fallback timed out."));
```

Comment:
You should rather fail with the appropriate token, so the user can continue the flow with a failed fingerprint